### PR TITLE
additional test on EVP_CipherUpdate/Final

### PR DESCRIPTION
**Pull request by kojo1**

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
...
+    (!defined(NO_AES) && defined(HAVE_AES_CBC) && defined(WOLFSSL_AES_128)) 
+static void binary_dump(void *ptr, int size)
+{
+    #ifdef WOLFSSL_EVP_PRINT
+   	int i = 0;
+	unsigned char *p = (unsigned char *) ptr;
+
+	printf("{");
+	while((p != NULL) && (i < size)) {
+		if((i % 8) == 0) {
```
Comment (cconlon) [inline tests/api.c:32081]: Please remove hard tabs in this function.

File: tests/api.c [hunk 2]
```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
...
+}
+
+static int last_val = 0x0f;
+
+static int check_result(unsigned char *data, int len)
+{
+	int i;
+	
+	for( ; len; ) {
+		last_val = (last_val + 1) % 16;
```
Comment (cconlon) [inline tests/api.c:32102]: Please remove hard tabs in this function.

File: tests/api.c [hunk 3]
```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
...
+static int w_offset;
+
+static void init_offset()
+{
+    r_offset = 0;
+    w_offset = 0;
+}
+static void get_record(unsigned char *data, unsigned char *buf, int len)
+{
+    memcpy(buf, data+r_offset, len);
```
Comment (cconlon) [inline tests/api.c:32121]: Please use XMEMCPY here instead of memcpy

File: tests/api.c [hunk 4]
```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
...
+    r_offset += len;
+}
+
+static void set_record(unsigned char *data, unsigned char *buf, int len)
+{
+    memcpy(data+w_offset, buf, len);
```
Comment (cconlon) [inline tests/api.c:32127]: Please use XMEMCPY here instead of memcpy

File: tests/api.c [hunk 5]
```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
...
+    int i, j;
+    unsigned char *p = plain;
+
+    #define BLOCKSZ 16
+
+    for(i=0; i<(rec/BLOCKSZ); i++){
+        for(j=0; j<BLOCKSZ; j++)
+            *p++ = (i % 16);
+    }
+    //binary_dump(plain, rec);
```
Comment (cconlon) [inline tests/api.c:32142]: Should we remove this commented out line?

File: tests/api.c [hunk 6]
```diff
@@ -32067,6 +32067,217 @@ tgZl96bcAGdru8OpQYP7x/rI4h5+rwA/kwIBAg==\n\
...
+    int test_drive1[] = {8, 3, 5, 512, 8, 3, 8, 512, 0};
+    int test_drive2[] = {8, 3, 8, 512, 0};
+    int test_drive3[] = {512, 512, 504, 512, 512, 8, 512, 0};
+
+    int *test_drive[] = {test_drive1, test_drive2, test_drive3, NULL};
+    int test_drive_len[100];
+    int drive_len;
+
+    int ret = 0;
+	EVP_CIPHER_CTX *evp = NULL;
```
Comment (cconlon) [inline tests/api.c:32173]: Please remove hard tabs in this function, thanks.

**Review by cconlon — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 1 above)
Comment (kojo1) [reply tests/api.c:32081]: fixed

File: tests/api.c (hunk 2 above)
Comment (kojo1) [reply tests/api.c:32102]: fixed

File: tests/api.c (hunk 4 above)
Comment (kojo1) [reply tests/api.c:32127]: fixed

File: tests/api.c (hunk 5 above)
Comment (kojo1) [reply tests/api.c:32142]: fixed

**Review by ejohnstown — APPROVED** [review]

**Review by cconlon — APPROVED** [review]

**Pull request merged**